### PR TITLE
Configure circle

### PR DESCRIPTION
**Pull request by messerc**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,241 @@
+#
+#  Globality autogenerated CircleCI configuration
+#
+#  This file is auto generated with globality-build.
+#  You should not make any changes to this file manually
+#
+#  Any changes made to this file will be overwritten in the
+#  next version of the build.
+#
+#  See: http://github.com/globality-corp/globality-build
```
Comment (cdm-ium) [inline .circleci/config.yml]: Let's look at the anatomy of this file. There are basically 3 seconds:
- define variables
- declare jobs
- establish the mapping between a commit to a branch/tag and a collection of jobs

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+#
+#
+
+defaults: &defaults
```
Comment (cdm-ium) [inline .circleci/config.yml]: Here is where we declare the variables necessary for the build.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        EXTRA_INDEX_URL: "InjectedDuringRuntime"
+        AWS_ECR_DOMAIN: "InjectedDuringRuntime"
+        JFROG_AUTH: "InjectedDuringRuntime"
+whitelist: &whitelist
+  paths:
+    .
+
+version: 2
+
+jobs:
```
Comment (cdm-ium) [inline .circleci/config.yml]: Here is where we declare the jobs. Each job is a straightfoward collection of shell commands, e.g. some yarn command to call into the commands defined in `package.json` in the app.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - setup_remote_docker
+      - run:
+          name: Deploy
+          command: |
+            eval $(aws ecr get-login --no-include-email)
+            mv Dockerfile.template Dockerfile
+            docker build -t ${AWS_ECR_DOMAIN}/truman:${CIRCLE_SHA1} .
```
Comment (cdm-ium) [inline .circleci/config.yml]: @messerc  update the name here. You can pick a code name for the poco sample service and swap it out with `truman`.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+    steps:
+      - attach_workspace:
+          at: ~/repo
+
+      - run:
+          name: Lint code
+          command: yarn lint
+
+  deploy_dev:
```
Comment (cdm-ium) [inline .circleci/config.yml]: Actually we should delete this job altogether as we don't want to deploy this.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                --sha ${CIRCLE_SHA1} \
+                --cluster "$(http -b http://color.staging.globality.io/newfrontend)"
+
+
+workflows:
+  version: 2
+
+  build-and-release:
+    jobs:
+      - checkout:
```
Comment (cdm-ium) [inline .circleci/config.yml]: If you read through this you should see that the `checkout`, `test` `lint` and `build` jobs should happen on every branch.

File: package.json [hunk 7]
```diff
@@ -17,7 +17,9 @@
   },
   "scripts": {
     "start": "yarn run serve",
-    "serve": "gi-webapp-run serve --port 5000"
+    "serve": "gi-webapp-run serve --port 5000",
+    "lint": "gi-webapp-run lint",
```
Comment (messerc) [inline package.json]: continue to leverage `gi-webapp-build-tools` to handle our linting and testing commands

**Comment by jameschung2019** [comment]
Hey there, please close this PR or merge it.   It will be auto-closed in 48 hours if not acted upon.  If you decide to merge it, comment and take ownership to coordinate, schedule, and get it done.  Thanks!

**Comment by jameschung2019** [comment]
This PR is being closed per CTO leadership's decision.  All PRs older than 1/1/2020 is included.

**Pull request closed without merging**